### ISSUE TITLE
`circe-split-line-length` not obeyed when line contains no whitespace

### ISSUE THREAD

**Issue by TaylanUB** [jorgenschaefer/circe#269]
When I post a very long string with no whitespace in it, Circe doesn't split it at all.

**Comment by wasamasa** [comment]
I suspect this is by design as it currently splits by whitespace.  After all, it would look weird if your lengthy message got trunc

**Comment by jorgenschaefer** [comment]
Yeah, Circe should really split long lines hard at a given line length. It currently uses `fill-region`. I suspect we can't re-use that anymore. (We should pass `NOSQUEEZE` anyhow …)
